environment.yml

```diff
@@ -48,7 +48,7 @@ dependencies:
   - nbconvert
   - nbsphinx>=0.7
   - nomkl
-  - pyarrow>=0.12 # must pin again otherwise strange things happen
+  - pyarrow>=0.12, <=0.15.1 # must pin again otherwise strange things happen
```

Comment:
really?

Reply:
This I plan to revert. It seems like tests will pass regardless the pyarrow version

Reply:
kk great

Reply:
Nevermind, looks like the `pyarrow` version actually does matter

Now that `pyspark` resolves to `2.4.6`, `pyarrow` needs to be `<=0.14.1`, otherwise UDF tests (also some temporal-related tests) fail (best guess as to why: https://github.com/ibis-project/ibis/pull/2216#discussion_r430687844).

I added a pin, but this time to the individual backend .yml files `pyspark.yml` and `spark.yml` rather than in the global `environment.yml`

--

From testing various `pyarrow` version locally:
| pyspark  | pyarrow | PySpark and Spark backend UDF tests |
| ------------- | ------------- | ------------- |
| 2.4.6  | 0.13.0 | ✅ All pass |
| 2.4.6  | 0.14.0 | ✅ All pass |
| 2.4.6  | 0.14.1 | ✅ All pass |
| 2.4.6  | 0.15.0 |  ❌ All fail |
| 2.4.6  | 0.15.1 |  ❌ All fail |
| 2.4.6  | 1.0.1 |  ❌ All fail |

Reply:
Yes this is unexpected. I think once we are on Spark 3.0 we can lift this version ping on pyarrow. (Haven't tested but I think Spark 3.0 works with PyArrow 1.0+)